src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift

```diff
@@ -0,0 +1,320 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+
+protocol PPAnalyticsSubmitting {
+	/// Triggers the submission of all collected analytics data. Only if all checks success, the submission is done. Otherwise, the submission is aborted. In all cases, NO completion is called.
+	func triggerSubmitData()
+
+	#if !RELEASE
+	/// ONLY FOR TESTING. Triggers the submission of all collected analytics data. Only if all checks success, the submission is done. Otherwise, the submission is aborted. The completion calls are passed through to test the component.
+	func triggerSubmitData(ppacToken: PPACToken?, completion: ((Result<Void, PPASError>) -> Void)?)
+	#endif
+}
+
+final class PPAnalyticsSubmitter: PPAnalyticsSubmitting {
+
+	// MARK: - Init
+
+	init(
+		store: Store,
+		client: Client,
+		appConfig: AppConfigurationProviding
+	) {
+		self.store = store
+		self.client = client
+		self.configurationProvider = appConfig
+	}
+
+	// MARK: - Protocol PPAnalyticsSubmitting
+
+	func triggerSubmitData() {
+		// 0. Check if user has given his consent to collect data
+		if userDeclinedAnalyticsCollectionConsent {
+			Log.warning("Analytics submission abord due to missing users consent", log: .ppa)
+			return
+		}
+
+		configurationProvider.appConfiguration().sink { [weak self] configuration in
+
+			guard let self = self else {
+				Log.warning("Analytics submission abord due fail at creating strong self", log: .ppa)
+				return
+			}
+
+			// 1. Check configuration parameter
+			if Double.random(in: 0...1) > configuration.privacyPreservingAnalyticsParameters.common.probabilityToSubmit {
+				Log.warning("Analytics submission abord due to randomness", log: .ppa)
+				return
+			}
```

Comment:
Yes, the case could happen that after 24 hours the submission is prevented and when then no data is collected, the latest next submission is after 24 hours again. Perhaps then the submission is prevented again....Honestly, I don't know how many times this could happen. 
But it just implemented in the order  as the tech spec defines. I will ask Max here.

Reply:
True, the order doesn't change anything. I guess I've had a nested approach in mind which defined the earliest & latest submission constraints. Let's stick to this then for now.